packages/cli/src/environments.ee/source-control/source-control.service.ee.ts

```diff
@@ -605,16 +633,21 @@ export class SourceControlService {
 		});
 
 		wfMissingInLocal.forEach((item) => {
-			sourceControlledFiles.push({
-				id: item.id,
-				name: item.name ?? 'Workflow',
-				type: 'workflow',
-				status: options.direction === 'push' ? 'deleted' : 'created',
-				location: options.direction === 'push' ? 'local' : 'remote',
-				conflict: false,
-				file: item.filename,
-				updatedAt: item.updatedAt ?? new Date().toISOString(),
-			});
+			// If we have out of scope workflows, these are workflows, that are not
+			// visible locally, but exists locally but are available in remote
+			// we skip them and hide them from deletion from the user.
+			if (!outOfScopeWF.some((outOfScope) => outOfScope.id === item.id)) {
```

Comment:
maybe it would be clearer to filter the workflow out in the filtering method here: https://github.com/n8n-io/n8n/pull/15404/files#diff-5360ad12c31a3065cde77f556ded52047c0e4311a2fde1a3822cd280d911f3d9R592